znnl/loss_functions/contrastive_infonce_loss.py

```diff
@@ -0,0 +1,145 @@
+"""
+ZnNL: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+"""
+
+import jax.numpy as np
+
+from znnl.loss_functions.contrastive_loss import ContrastiveLoss
+
+
+class ContrastiveInfoNCELoss(ContrastiveLoss):
+    """
+    Class for the Contrastive InfoNCE Loss.
+
+    The InfoNCE loss is a type of contrastive loss function.
+    It is used to train neural networks by maximizing the mutual information between
+    the input and the output of the network.
+
+    The InfoNCE loss is an implementation of a contrastive loss function using a
+    repulsive and attractive potential in one term.
+    The loss is scaled by a temperature parameter.
+    """
+
+    def __init__(
+        self,
+        temperature: float = 1.0,
+    ):
+        """
+        Constructor for the Contrastive Loss class.
+
+        Parameters
+        ----------
+        temperature : float
+                The temperature parameter for the InfoNCE loss.
+        """
+
+        super().__init__()
+        self.temperature = temperature
+
+    def compute_losses(self, inputs: np.ndarray, targets: np.ndarray):
```

Comment:
Can you add output type hints here and everywhere

Reply:
Added type hints everywhere in 
[7d57017](https://github.com/zincware/ZnNL/pull/116/commits/7d570172057c05a09394712e0b1ff942ceb5611f)